docs/examples/example_oosm_algorithm.py

```diff
@@ -0,0 +1,405 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+"""
+===================================================
+Algorithm dealing with Out-of-Sequence measurements
+===================================================
+"""
+
+# %%
+# In previous examples, we have shown how to deal with out-of-sequence measurements (OOSM) by
+# using a fixed time-delay, or a time buffer. This example focuses on a different approach to deal
+# with OOSM.
+#
+# As shown in literature (e.g., [1]_) there are multiple approaches on how to deal with OOS measurements,
+# spanning from simply ignoring them, assuming that the fraction of these is small and it will not,
+# significally, impact the quality of the tracking, iterate over last :math:`\ell` measurements with
+# a fixed lag (see what it is called Algorithm A and B in [2]_ and [3]_, see also the previous examples)
+# or, like in this example, you can include any OOSM and re-process the measurement using
+# inverse-time dynamics creating pseudo-measurements.
+#
+# This approach is called in literature as Algorithm C (from [2]_, [3]_) and it will be refered as such later on.
+#
+# To explain how it works, we consider a single target scenario.
+# This algorithm deals with the presence of delayed measurements by using the predicted dynamics of the target
+# to go back in time from the actual arrival time and obtain a pseudo-measurement obtained from
+# simulating the detection at the "expected" scan time.
+# In this manner we can reconstruct the measurements chain and not discard any information, as well,
+# we don't need to store a large fix-lag distribution of data (as can happen with other algorithms).
+#
+# In this example, we focus our efforts in showing how to use the algorithm C in a multi-target scenario with clutter.
+# We simulate two sensors obtaining scans of two objects travelling with a nearly constant velocity transition model
+# and with some level of clutter, at specific timesteps we insert a delay in one of the scan from the sensor and we
+# employ Algorithm C to process in the correct way the chain of measurements.
+#
```

Comment:
```suggestion
# As shown in literature [1]_ there are multiple approaches on how to deal with OOS measurements.
# Simply ignoring them, assuming that the fraction of these is small and it will not
# significantly impact the quality of the tracking; iterating over last :math:`\ell` measurements with
# a fixed lag (see what it is called Algorithm A and B in [2]_ and [3]_, and the other OOSM examples);
# or, like in this example, you can include any OOSM and re-process the measurement using
# inverse-time dynamics creating pseudo-measurements.
#
# This approach is known in literature as Algorithm C [2]_ [3]_ and it will be referred to as such later on.
#
# To explain how it works, we consider a single target scenario.
# This algorithm deals with the presence of delayed measurements by using the predicted dynamics of the target
# to go back in time from the actual arrival time and obtain a pseudo-measurement from
# simulating the detection at the "expected" scan time.
# In this manner we can reconstruct the measurements chain without discarding any information, and
# we don't need to store a large fix-lag distribution of data (as can happen with other algorithms).
#
# In this example, we focus our efforts in showing how to use the algorithm C in a multi-target scenario with clutter.
# We simulate two sensors obtaining scans of two objects travelling with a nearly constant velocity transition model
# and with some level of clutter. At specific timesteps we insert a delay in one of the scans from the sensor and we
# employ Algorithm C to process in the correct way the chain of measurements.
#
```